"Show Techsupport" HLD

kerry-meyer opened the pull request:
This pull request is being opened for review, revision as needed, and merging of the "Show Techsupport" HLD at:

https://github.com/project-arlo/SONiC/blob/show-techsupport-hld/doc/mgmt/SONiC%20Management%20Framework%20Show%20Techsupport%20HLD.md

kerry-meyer committed:
Uploading the "Show Techsupport" HLD for review.

jeffyin-dell commented on doc/mgmt/SONiC Management Framework Show Techsupport HLD.md:
```diff
@@ -0,0 +1,189 @@
+
+module: sonic-show-techsupport
+
+  rpcs:
```
This is not going to render correctly. Please use triple-backticks (```) to enclose this section to make it appear as a monospaced code block.

jeffyin-dell commented on doc/mgmt/SONiC Management Framework Show Techsupport HLD.md:
```diff
@@ -0,0 +1,189 @@
+N/A
+#### 3.6.2.2 Show Commands
+
+show techsupport [since <DateTime>] — Gather information for troubleshooting. Display the name of a file containing the resulting group of collected information items in a compressed "tar" file.
```
Please enclose CLI examples and commands in backticks (`) to render them as monospaced text.

kerry-meyer replied:
Done. (Please see the new revision.)

jeffyin-dell commented on doc/mgmt/SONiC Management Framework Show Techsupport HLD.md:
```diff
@@ -0,0 +1,189 @@
+Command Mode: User EXEC
+
+Example:
+
```
Enclose this section in triple-backticks (```)

kerry-meyer replied:
Done. (Please see the new revision.)

jeffyin-dell commented on doc/mgmt/SONiC Management Framework Show Techsupport HLD.md:
```diff
@@ -0,0 +1,189 @@
+Refer to section 1.1.3
+
+# 9 Unit Test
+- Basic command execution:
```
This would look better as a table

kerry-meyer replied:
Done. (Please see the new revision.)

jeffyin-dell commented on doc/mgmt/SONiC Management Framework Show Techsupport HLD.md:
```diff
@@ -0,0 +1,189 @@
+N/A
+
+# 6 Serviceability and Debug
+Logging, counters, stats, trace considerations. Please make sure you have incorporated the debugging framework feature. e.g., ensure your code registers with the debugging framework and add your dump routines for any debug info you want to be collected.
```
Please replace the default text with content specific to your feature, or mark it as N/A

kerry-meyer replied:
Done. (Please see the new revision.)

jeffyin-dell reviewed (changes requested):
Requesting formatting changes

kerry-meyer committed:
Uploading a revised version to address comments from Jeff Yin

This revision incorporates formatting changes to improve rendering and readability of the document.

jeffyin-dell commented on doc/mgmt/SONiC Management Framework Show Techsupport HLD.md:
```diff
@@ -0,0 +1,198 @@
+       |  +---w date?   yang:date-and-time
+       +--ro output
+          +--ro output-filename?   string
+
```
You need to terminate the code block here with triple-backticks (```)

jeffyin-dell commented on doc/mgmt/SONiC Management Framework Show Techsupport HLD.md:
````diff
@@ -0,0 +1,198 @@
+N/A
+#### 3.6.2.2 Show Commands
+
+```
````
Remove this line of ```

kerry-meyer replied:
Done

jeffyin-dell commented on doc/mgmt/SONiC Management Framework Show Techsupport HLD.md:
````diff
@@ -0,0 +1,198 @@
+```
+Command syntax summary:
+
+show techsupport [since <DateTime\>]
````
enclose this line in single backticks to make it monospaced. Like this:
`show techsupport [since <DateTime>]`

jeffyin-dell replied:
Still need to fix this..

jeffyin-dell commented on doc/mgmt/SONiC Management Framework Show Techsupport HLD.md:
```diff
@@ -0,0 +1,198 @@
+#### 3.6.2.4 IS-CLI Compliance
+N/A (TBD)
+
+The following table maps SONIC CLI commands to corresponding IS-CLI commands. The compliance column identifies how the command comply to the IS-CLI syntax:
```
Delete this section up to and including the table, and just explain that there's no "IS-CLI" equivalent, at least not the way it works in SONiC. (The "IS-CLI" version dumps it all to the buffer, whereas this one dumps to a compressed file)

kerry-meyer replied:
Done.

jeffyin-dell commented on doc/mgmt/SONiC Management Framework Show Techsupport HLD.md:
````diff
@@ -0,0 +1,198 @@
+```
+sonic# show techsupport
+
+Output stored in:  /var/dump/sonic_dump_sonic_20191008_082312.tar.gz
````
You should note that getting to this file _will_ require that the user drop out of the management-framework CLI and into the host's Bash shell in order to get to it, copy it, transfer it, etc.

jeffyin-dell replied:
This is a limitation of the CLI; we will likely add file manipulation commands in a future release...

kerry-meyer replied:
I added a note to that effect in section 3.6.1 (IS-CLI deviations). I have added a reference to that note here, along with a statement that it is desirable for a Management Framework based file transfer capability to be added in a future release to make this feature more convenient.

kerry-meyer committed:
This version addresses follow-up comments from Jeff Yin.

Add formatting fixes and update content in several sections. Add documentation on the limitations of the initial implementation and desirable supplementary file transfer capability in a future release to make the current implementation fully compatible with a management framework based client interface.

kerry-meyer committed:
Added backtick enclosure for the command syntax specification.

This comment was missed in previous revisions. This version provides the desired mono-spacing for the syntax specification.

kerry-meyer commented:
I have fixed the missing monospace marking for the syntax specification. Sorry that I missed it earlier.

jeffyin-dell commented:
@dutta-partha would you like to provide a review? If not, will merge this tomorrow.

kerry-meyer committed:
Correct a duplicated unit test case to specify the correct test description

The "Unit Test" section of the document is intended to specify two invalid "date" cases. One is an invalid date. The other (applicable only to the CLI client interface) is typing the "date" sub-command/keyword without following it with text specifying a Date/time value ("no" date string after entering the "date" sub-command mode). The document previously contained two instances of the first of these two cases and was missing the second one.

I have corrected this mistake by editing the second copy of the "invalid date" case to specify that the testing should cover the "no date" case.

sreenath-kada commented:
Hi 

Few comments:

In 2.2 - Please add details of what is the formatting/packing that will be done and how the user has to reformat/unpack them to retrieve the original files

In 3.1 - Please add details on how the unique name is determined

In 3.1 - Please elaborate what other items are considered for since option apart from logs

In 6 - Why is there no serviceability built in, what is this command fails? How will the user know what is wrong?

Any thoughts on generating show tech-support per component?

In section 1.1.3 - The scalability requirements can be a little more explicit. It can state the approximate time taken on a CPU of specific power rather than use the term "high end"

It would be good to have an example of "show tech-support" and "since" flag outputs.
The REST and gNMI interface examples to trigger the tech support may also be added to the document.

Would be good to have a simple flow diagram for section 3.1.

Regards
Sreenath
Broadcom Customer Support

kerry-meyer commented:
Internal Use - Confidential

Hi Sreenath,

Thank you for your feedback on the Show Techsupport HLD document. I am working now on adding updates to the document based on your comments There are a couple of issues raised in your comments that my managers are considering. We will get back to you on these items soon.

                      Kerry


From: Sreenath Reddy Kada <notifications@github.com>
Sent: Monday, November 18, 2019 2:55 PM
To: project-arlo/SONiC
Cc: Meyer, Kerry; Assign
Subject: Re: [project-arlo/SONiC] "Show Techsupport" HLD (#43)


[EXTERNAL EMAIL]

Hi

Few comments:

In 2.2 - Please add details of what is the formatting/packing that will be done and how the user has to reformat/unpack them to retrieve the original files

In 3.1 - Please add details on how the unique name is determined

In 3.1 - Please elaborate what other items are considered for since option apart from logs

In 6 - Why is there no serviceability built in, what is this command fails? How will the user know what is wrong?

Any thoughts on generating show tech-support per component?

In section 1.1.3 - The scalability requirements can be a little more explicit. It can state the approximate time taken on a CPU of specific power rather than use the term "high end"

It would be good to have an example of "show tech-support" and "since" flag outputs.
The REST and gNMI interface examples to trigger the tech support may also be added to the document.

Would be good to have a simple flow diagram for section 3.1.

Regards
Sreenath
Broadcom Customer Support

-
You are receiving this because you were assigned.
Reply to this email directly, view it on GitHub<https://github.com/project-arlo/SONiC/pull/43?email_source=notifications&email_token=AING7LYQJCWE4S5622EOSU3QUMMMDA5CNFSM4JJMZBE2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEEMGWVI#issuecomment-555248469>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AING7L3GDFII6QQXPB2535DQUMMMDANCNFSM4JJMZBEQ>.

kerry-meyer committed:
Provide changes to address review comments: Add requested additional descriptions.

The set of changes included in this new version are intended to address the majority of the comments posted by Sreenath Kada for the pending merge request for the "show tech-support" HLD.

Additional changes have been made to provide supplementary information and explanations, to make corrections for changes made in the design after the initial posting of the HLD, and to correct grammar, punctuation, and text highlighting.

kerry-meyer commented:
A new version of the specification has been posted in response to the comments from Sreenath. Please see inline below for specific responses the comments that were posted. (Responses are preceded by "KM".)

> 
> 
> Hi
> 
> Few comments:
> 
> In 2.2 - Please add details of what is the formatting/packing that will be done and how the user has to reformat/unpack them to retrieve the original files

KM: I have added these details in section 2.2 (Functional Description). (Please note that this is the same as for the already existing "show tech-support" command. Documentation on this base functionality is available at:   https://github.com/Azure/sonic-utilities/blob/master/doc/Command-Reference.md#troubleshooting-commands
> 
> In 3.1 - Please add details on how the unique name is determined
> 
KM: I have added this information in section 3.6.2.2 (Show Commands)

> In 3.1 - Please elaborate what other items are considered for since option apart from logs
> 
KM: The date/time specified via the "since" option is only used for log and core files. I have specified this in section 3.6.2.2 (Show Commands).
> In 6 - Why is there no serviceability built in, what is this command fails? How will the user know what is wrong?
> 
KM: Any errors encountered during execution of the command that prevent retrieval of the technical support information are posted in an output message presented to the client interface. I have added a statement to this effect in section 6 (Serviceability and Debug).
> Any thoughts on generating show tech-support per component?
> 
KM: This is beyond the scope of the feature: This feature only provides a Management Framework front end (enabling invocation via REST API, gNOI, or CLI) for the underlying "show tech-support" functionality.
> In section 1.1.3 - The scalability requirements can be a little more explicit. It can state the approximate time taken on a CPU of specific power rather than use the term "high end"
> 
KM: This feature is just a management framework "front end" for the already existing "show tech-support" functionality. It does not affect the performance characteristics of the command (latency, execution time, storage space). I have added wording to this effect in section 1.1.3 (Scalability Requirements). 
> It would be good to have an example of "show tech-support" and "since" flag outputs.
> The REST and gNMI interface examples to trigger the tech support may also be added to the document.
> 
KM: I have added the requested examples in section 3.2.2.2 (Show Commands).
> Would be good to have a simple flow diagram for section 3.1.
> 
The flow of control is the same as for other commands executed via the Management Framework infrastructure. Detailed descriptions of the flow of control are provided in the Management Framework specification: https://github.com/project-arlo/SONiC/blob/master/doc/mgmt/Management%20Framework.md 

> Regards
> Sreenath
> Broadcom Customer Support

kerry-meyer committed:
Added a reference in the Overview section to the document describing the underlying "show techsupport" feature.
